### PR TITLE
Enabling automated tests for MySQL & MariaDB

### PR DESCRIPTION
**Pull request by jinengandhi-intel**

### CODE REVIEW

**Comment by anjalirai-intel** [comment]
Should we also add sudo to libs/utils.pipeline to ssh command?

File: data/tests_with_attestation.yaml [hunk 1]
```diff
@@ -176,4 +176,42 @@ test_pytorch_default_with_debug:
...
+  distro: 4
+
+test_mariadb_with_attestation:
+  docker_image: mariadb mariadb:10.7
+  runtime_args_text: --datadir /var/run/test_db_encrypted
+  docker_flags: -v /var/run/test_db_encrypted:/var/run/test_db_encrypted
+  encryption_key: workloads/mariadb/base_image_helper/encryption_key
+  encrypted_files_path: /var/run/test_db_encrypted
+  pre_workload_setup: y
+  distro: 4
```
Comment (anjalirai-intel) [inline data/tests_with_attestation.yaml:207]: mariadb distro is 2

File: libs/utils.py [hunk 2]
```diff
@@ -121,6 +122,48 @@ def generate_local_image(workload_image):
...
+        run_subprocess(MARIADB_CHMOD, CURATED_APPS_PATH)
+    return init_result
+
+def encrypt_db(workload_name):
+    output = run_subprocess(eval(workload_name.upper()+"_TEST_ENCRYPTION_KEY"), CURATED_APPS_PATH)
+    output = run_subprocess(CLEANUP_ENCRYPTED_DB, CURATED_APPS_PATH)
+    encryption_output = run_subprocess(eval(workload_name.upper()+"_ENCRYPT_DB_CMD"), CURATED_APPS_PATH)
+
+def execute_pre_workload_setup(test_config_dict):
+    workload_name = test_config_dict["docker_image"].split()[0]
```
Comment (anjalirai-intel) [inline libs/utils.py:160]: get_workload_name can be used

File: data/constants.py [hunk 3]
```diff
@@ -53,3 +53,32 @@
...
+                            -v $PWD/workloads/mysql/test_db:/test_db \
+                            -e MYSQL_ALLOW_EMPTY_PASSWORD=true -e MYSQL_DATABASE=test_db mysql:8.0.32-debian \
+                            --datadir /test_db"
+STOP_TEST_DB_CMD      = f"docker stop init_test_db"
+MYSQL_TEST_ENCRYPTION_KEY    = f"dd if=/dev/urandom bs=16 count=1 > workloads/mysql/base_image_helper/encryption_key"
+
+CLEANUP_ENCRYPTED_DB   = f"sudo rm -rf /var/run/test_db_encrypted"
+MYSQL_ENCRYPT_DB_CMD         = f"sudo gramine-sgx-pf-crypt encrypt -w workloads/mysql/base_image_helper/encryption_key \
+                            -i workloads/mysql/test_db -o /var/run/test_db_encrypted"
+MYSQL_CLIENT_INSTALL_CMD = f"sudo apt-get -y install mysql-client"
```
Comment (anjalirai-intel) [inline data/constants.py:69]: "f" for all the strings may not be required

**Comment by jinengandhi-intel** [comment]
> Should we also add sudo to libs/utils.pipeline to ssh command?

It passed without adding any sudo: http://ba5sapp0350:8080/view/Curation%20App/job/jinen_curation_app_test/8/console

The final failure is because I was searching for mysql in the list of users which won't be present for MariaDb, have fixed it in the latest commit and now I only look for root.

**Review by anjalirai-intel — APPROVED** [review]

**Pull request merged**